chain/client/src/client.rs

```diff
@@ -320,6 +321,18 @@ impl ClientActor {
         if status == BlockStatus::Next || status == BlockStatus::Reorg {
             self.tx_pool.reconcile_block(&block);
         }
+
+        // Announce AccountId if client is becoming a validator soon.
+        // First check that we currently have an AccountId
+        if let Some(block_producer) = self.block_producer.as_ref() {
+            // TODO(MarX): Check if I'm becoming a validator
+            // TODO(MarX): Check if I haven't announced my account for next epoch
+            // self.runtime_adapter.get_epoch_offset(block.hash(), block.header.height + 1)?;
+            let account_id = block_producer.account_id.clone();
+            let signature = block_producer.signer.sign(account_id.as_bytes());
```

Comment:
hm, signature should be on account_id+peer_id may be? and possibly the current epoch
because this signature can be used by anyone right now

Reply:
To let the client be aware of the PeerId (in order to sign the message) I needed to:
1. add peer_id field to client (this is not desirable since client shouldn't be aware of anything network related).

Other solutions:
2. Extra communication between client and network to request/notify peer_id value.
3. Let the network sign on behalf of the client. Also not desirable to mess up functionalities from each layer.

I went with option 1 but can change it.